### PR TITLE
API - UnitTest - Get Address Location and Location Detail

### PR DESCRIPTION
**Pull request by at-haingo**
#### Related issues (Required)
- Task #221 

#### Description (Required)
- [x]  API - UnitTest - Get Address Location and Location Detail

### CODE REVIEW

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this PR</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">All test passed. (22/22)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt [hunk 1]
```diff
@@ -61,6 +63,56 @@ class GoogleApiTest {
...
+            assertThat(it.results?.size, `is`(3))
+            assertThat(item?.address, `is`("Nại Thịnh 5, Nại Hiên Đông, Sơn Trà, Đà Nẵng, Vietnam"))
+            assertThat(item?.placeId, `is`("ChIJ8bYoOR8YQjERMiVjLDsw3Kg"))
+
+            it.results!!.isNotEmpty()
+        }
+    }
+
+    @Test
+    fun `Given mock response -  When request getLocationDetail - Then ResultPlaceDetail`() {
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt]: `Then ResultPlaceDetail` change to `Then return...`

File: app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt [hunk 2]
```diff
@@ -61,6 +63,56 @@ class GoogleApiTest {
...
+
+        /* Then */
+        val request = server.takeRequest()
+        assertThat(request.method.toUpperCase(), `is`("GET"))
+        assertThat(request.requestUrl.queryParameterNames(), hasItem("latlng"))
+        assertThat(request.requestUrl.queryParameter("latlng"), `is`("16.087190, 108.232773"))
+
+        test.assertValue {
+            val item = it.results?.get(0)
+            assertThat(it.results?.size, `is`(3))
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt]: move this line to line 84

File: app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt [hunk 3]
```diff
@@ -61,6 +63,56 @@ class GoogleApiTest {
...
+    }
+
+    @Test
+    fun `Given mock response -  When request getAddressLocation - Then return response mutable list LocationAddress`() {
+        /* Given */
+        val test = TestObserver<Response<MutableList<LocationAddress>>>()
+        server.addResponseBody("getAddressLocation.json")
+
+        /* When */
+        restClient.getAddressLocation("16.087190, 108.232773").subscribe(test)
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt]: define a variable for this value `"16.087190, 108.232773"`

File: app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt [hunk 4]
```diff
@@ -61,6 +63,56 @@ class GoogleApiTest {
...
+    }
+
+    @Test
+    fun `Given mock response -  When request getLocationDetail - Then return ResultPlaceDetail`() {
+        /* Given */
+        val test = TestObserver<ResultPlaceDetail>()
+        server.addResponseBody("getLocationDetail.json")
+
+        /* When */
+        restClient.getLocationDetail("ChIJ8bYoOR8YQjERMiVjLDsw3Kg").subscribe(test)
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/api/GoogleApiTest.kt]: define variable for this value `ChIJ8bYoOR8YQjERMiVjLDsw3Kg`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

**Pull request merged**